### PR TITLE
feat(logs): Add grantRead to LogGroup construct

### LINKED ISSUE #21668: (CloudWatch LogGroup): (its missing a grantRead)
### Describe the feature

CloudWatch LogGroup CDK construct does have a grantWrite but not a grantRead, it would be great if it was added, there are too many actions to add to grant(IGrantable, actions) LogGroup

### Use Case

trying to restrict read access to a specific CloudWatch LogGroup

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.35.0

### Environment details (OS name and version, etc.)

Linux AL2

### PR DESCRIPTION
**Pull request by markmansur**
----

### Motivation
fixes #21668 

Adds `grantRead` method to `LogGroup` construct. Grant's the following IAM actions `logs:FilterLogEvents`, `logs:GetLogEvents` and `logs:GetLogRecord`.

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-logs/lib/log-group.ts [hunk 1]
```diff
@@ -169,6 +174,13 @@ abstract class LogGroupBase extends Resource implements ILogGroup {
     return this.grant(grantee, 'logs:CreateLogStream', 'logs:PutLogEvents');
   }
 
+  /**
+   * Give permissions to read events from this log group
+   */
+  public grantRead(grantee: iam.IGrantable) {
+    return this.grant(grantee, 'logs:FilterLogEvents', 'logs:GetLogEvents	', 'logs:GetLogRecord')
```
Comment (TheRealAmazonKendra) [inline packages/@aws-cdk/aws-logs/lib/log-group.ts:181]:
```suggestion
    return this.grant(grantee, 'logs:FilterLogEvents', 'logs:GetLogEvents', 'logs:GetLogRecord');
```

**Review by TheRealAmazonKendra — DISMISSED** [review]

File: packages/@aws-cdk/aws-logs/lib/log-group.ts (hunk 1 above)
Comment (markmansur) [reply packages/@aws-cdk/aws-logs/lib/log-group.ts]: Oops, fixed!

**Comment by markmansur** [comment]
> 

Do we really need an integ test for this? I think unit tests are more than sufficient here. I can add integ tests if needed..

**Comment by markmansur** [comment]
> The Pull Request Linter fails with the following errors:
> 
> ```
> ❌ Features must contain a change to an integration test file and the resulting snapshot.
> ```
> 
> PRs must pass status checks before we can provide a meaningful review.

Do we really need integ tests for this? I think unit tests are more than enough here. Thoughts? I can add an integ test if needed..

**Comment by TheRealAmazonKendra** [comment]
> > The Pull Request Linter fails with the following errors:
> > ```
> > ❌ Features must contain a change to an integration test file and the resulting snapshot.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > PRs must pass status checks before we can provide a meaningful review.
> 
> Do we really need integ tests for this? I think unit tests are more than enough here. Thoughts? I can add an integ test if needed..
> 
> I'm a little confused about what the integ tests in `aws-logs` (and in general throughout the cdk codebase) are doing. It seems they are just snapshot tests? Does this count as a integ test? To me, an integ test would be actually deployed with CloudFormation in a real account and unit tests should be doing the snapshots. Am I wrong here?
> 
> I see in the contributing guide it mentions that the integ tests run `cdk deploy`. Is this actually deploying to a test aws account? or is it just a way to verify the template is valid?
> 
> @TheRealAmazonKendra your input here would be great :)

Yes, we do nee an integ test for this. When we run the integ tests and we know there's been a change, run with `--update-on-failed`. This will create the snapshots as long as the resources successfully deploy. I suggest you take a read through out [INTEGRATION_TEST](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md) file for more context and guidance.

**Pull request closed without merging**